tree/20_cfe_basics/time_lib.cf

```diff
@@ -0,0 +1,203 @@
+#####################################################################################
+# Copyright 2015 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @agent_version >=3.6
+#
+# This collection of bundles manage time related calculations
+#
+# There are 2 kind of scheduling: purely periodic and calendar based
+#
+# There are 6 cases for error handling, based on:
+#  - 2 miss -> 0 job, 1 job, 2 jobs
+#  - agent overlap -> 2 jobs, 1 job
+# For which we gave 6 names:
+# - stateless:     2miss->0job & 2run->2job
+# - no_dups:       2miss->0job & 2run->1job
+# - at_least_once: 2miss->1job & 2run->2job
+# - catch_up:      2miss->1job & 2run->1job
+# - no_miss:       2miss->2job & 2run->2job
+# - exact_count:   2miss->2job & 2run->1job
+
+
+# Stores now(), you can change now()'s result if you want to run tests by calling this bundle with a parameter
+bundle agent date(test_timestamp)
```

Comment:
i fear this name is too generic, and can overlap with user defined code (or even one day there may be a date method in the agent).
Could you give a more "unique" name?